pkg/providers/vsphere/vmprovider_vm_utils.go

```diff
@@ -48,6 +48,25 @@ func errToConditionReasonAndMessage(err error) (string, string) {
 	}
 }
 
+func errorMessageFromTaskInfoWithKey(taskInfo *vimtypes.TaskInfo, key string) string {
```

Comment:
Just double checking, are there any helper messages for this in govmomi?

Reply:
Saw old code doing similar things here https://github.com/vmware-tanzu/vm-operator/blob/4578f25e5bd528fdcc6c1d1f6aa32166944e748e/pkg/providers/vsphere/resources/vm.go#L241-L250

Maybe could be combined.

Reply:
The above code in [vm-operator/pkg/providers/vsphere/resources/vm.go](https://github.com/vmware-tanzu/vm-operator/blob/4578f25e5bd528fdcc6c1d1f6aa32166944e748e/pkg/providers/vsphere/resources/vm.go#L241-L250) stops at the Fault itself. In snapshot's case, we need to go a little deeper to see if it actually failed due to crossing a CSI Snapshot while attempting the revert.